client/my-sites/plans-v2/products-grid-alt/index.tsx

```diff
@@ -33,76 +27,22 @@ import type { Duration, PurchaseCallback, QueryArgs, SelectorProduct } from '../
  */
 import './style.scss';
 
-// Map over all plan slugs and convert them to SelectorProduct types.
-const getPlansToDisplay = ( {
-	duration,
-	currentPlanSlug,
-}: {
-	duration: Duration;
-	currentPlanSlug: string | null;
-} ): SelectorProduct[] => {
-	const currentPlanTerms = currentPlanSlug
-		? [ getMonthlyPlanByYearly( currentPlanSlug ), getYearlyPlanByMonthly( currentPlanSlug ) ]
-		: [];
-
-	const currentPlanOptions = currentPlanSlug ? getAllOptionsFromSlug( currentPlanSlug ) : [];
-
-	const plansToDisplay = SELECTOR_PLANS_ALT.map( slugToSelectorProduct )
-		// Remove plans that don't fit the filters or have invalid data.
-		.filter(
-			( product: SelectorProduct | null ): product is SelectorProduct =>
-				!! product &&
-				product.term === duration &&
-				// Don't include a plan the user already owns, regardless of the term
-				! currentPlanTerms.includes( product.productSlug ) &&
-				// Don't include a plan option if the user already owns a related option
-				! currentPlanOptions.includes( product.productSlug )
-		)
-		.map( ( product: SelectorProduct ) => ( {
-			...product,
-			description: getJetpackDescriptionWithOptions( product ),
-		} ) );
-
-	if (
-		currentPlanSlug &&
-		( JETPACK_LEGACY_PLANS.includes( currentPlanSlug ) ||
-			SELECTOR_PLANS_ALT.includes( currentPlanSlug ) )
-	) {
-		const currentPlanSelectorProduct = slugToSelectorProduct( currentPlanSlug );
-		if ( currentPlanSelectorProduct ) {
-			return [ currentPlanSelectorProduct, ...plansToDisplay ];
-		}
-	}
-
-	return plansToDisplay;
-};
-
-const getProductsToDisplay = ( {
-	duration,
-	availableProducts,
-	purchasedProducts,
-	includedInPlanProducts,
-}: {
-	duration: Duration;
-	availableProducts: ( SelectorProduct | null )[];
-	purchasedProducts: ( SelectorProduct | null )[];
-	includedInPlanProducts: ( SelectorProduct | null )[];
-} ) => {
-	// Products that have not been directly purchased must honor the current filter
-	// selection since they exist in both monthly and yearly version.
-	const filteredProducts = [ ...includedInPlanProducts, ...availableProducts ]
-		// Remove products that don't match the selected duration
-		.filter( ( product ): product is SelectorProduct => product?.term === duration );
-	return (
-		[ ...purchasedProducts, ...filteredProducts ]
-			// Make sure we don't allow any null or invalid products
-			.filter( ( product ): product is SelectorProduct => !! product )
-			.map( ( product ) => ( {
-				...product,
-				description: getJetpackDescriptionWithOptions( product as SelectorProduct ),
-			} ) )
-	);
-};
+const paidUpgradesDisplayOrder = [
```

Comment:
What if we transform this into a object that contains the position of every product in the grid. It would look like:
```
{
    PlanConstants.PLAN_JETPACK_SECURITY_DAILY: 1,
	PlanConstants.PLAN_JETPACK_SECURITY_DAILY_MONTHLY: 1,
    ProductConstants.PRODUCT_JETPACK_BACKUP_DAILY: 2,
	ProductConstants.PRODUCT_JETPACK_BACKUP_DAILY_MONTHLY: 2,
...
}
```
Then, we can iterate over the list of things we are going to show and sort them according to this.